### ISSUE TITLE
Documentation for structured demangling

### ISSUE THREAD

**Issue by actioninja** [gimli-rs/cpp_demangle#285]
I need to get the resulting AST from a demangle for usage in reverse engineering. I'm struggling to figure out how to actually do this, since no API surface seems to actually return the AST tree and there's no working example of how to do so.

**Comment by khuey** [comment]
We don't currently expose the full AST. We could add something to do that, but I wouldn't document it beyond "here's the cxxabi spec, good luck".

**Comment by fitzgen** [comment]
The AST would likely have breaking changes ~every release, which seems really unfortunate for basically all users who wouldn't be using the AST.

**Comment by khuey** [comment]
Yes, I think we'd do it with an off by default feature that says "if you turn this on semver goes out the window and you get zero compatibility guarantees"
